### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by kychua**
Future work:
- tools for documentation
- process for writing documentation
-  documentation pitfalls/anti-patterns or how not to document

### CODE REVIEW

**Comment by damithc** [comment]
@kychua can you get this peer reviewed by someone?

**Comment by kychua** [comment]
@wkurniawan07 can you please help me review this? Thanks!

**Comment by wkurniawan07** [comment]
Sure! Give me just a moment.

File: contents/projectManagement/documentation.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+> [Advice for Computer Science College Students](https://www.joelonsoftware.com/2005/01/02/advice-for-computer-science-college-students/), by Joel Spolsky
+
+## Why Write Documentation
+
+There are many benefits to writing good documentation, such as the following:
+
+* Make things easier for yourself: when you look back at your code a few months later, well-written documentation can make things a lot easier by explaining what you had been thinking when you were writing the code, when you find yourself confused by your own work
+* Improve your code: documentation forces you to think carefully about what you are trying to do and why you have chosen to implement it in a particular way
+* Encourage others to contribute to your software: by documenting how to contribute, it makes easier for others to do so, so they are more likely to help out
+* Encourage others to use your software: ["If it isn't documented, it doesn't exist."](http://www.mikepope.com/blog/DisplayBlog.aspx?permalink=1680)
```
Comment (wkurniawan07) [inline contents/projectManagement/documentation.md]: Could be better if formatted this way:
* **Main point**: explanation.

File: contents/projectManagement/documentation.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+In general, written documentation should use:
+
+* grammatically correct language
+* [simple language](https://wiki.zephyrproject.org/view/Documentation_style_guide#Simple_English): your documentation might be read by non-native speakers and use of idioms or flowery language can make it difficult for readers to understand
+* markup: use formatting such as colors and italics and structural elements like admonition blocks to highlight important points
+* realistic examples
+
+Futhermore, you should always:
+
+* [test](http://idratherbewriting.com/2015/07/07/testing-your-instructions/) your instructions
```
Comment (wkurniawan07) [inline contents/projectManagement/documentation.md]: The link should wrap the entire "test your instructions" phrase.

File: contents/projectManagement/documentation.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+[This](http://www.bluemangolearning.com/software-documentation/) provides useful general tips for writing documentation and 
+[this guide](https://jacobian.org/writing/technical-style/) provides more tips on writing well. 
+
+#### README
+
+A README serves as an introduction to your project, and is often the first thing people see when encountering your project.
+A good README should convince users to use your product or contribute to your code, by providing enough information to make it easy for them to get started.
+For a good guide on how to write a README, check out [this guide](http://www.writethedocs.org/guide/writing/beginners-guide-to-docs/).
+Some suggest that READMEs should be written before you start coding, as it gives you an overview of your project and what you need to do.
+This is known as [Readme Driven Development](http://tom.preston-werner.com/2010/08/23/readme-driven-development.html).
```
Comment (wkurniawan07) [inline contents/projectManagement/documentation.md]: Structure it this way?
```markdown
A README ...
A good README...

For a good guide on ...
- Some suggest that ... This is known as ...
```

File: contents/projectManagement/documentation.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+A user guide teaches users how to use your software.
+User guides can include quickstart guides, reference materials, tutorials, cookbooks/recipes and more.
+[Here](https://jacobian.org/writing/what-to-write/) is a useful guide on how to tackle different types of user documentation.
+[This](http://stevelosh.com/blog/2013/09/teach-dont-tell/) points out the problems with common approaches to documentation and builds on the previous article.
+
+#### Developer Guide
+
+Developer guides teach new developers how to contribute to the code base.
+They should give an overview of the code and an explanation of the design, as well as include instructions for setting up, making changes and deploying.
+See [this](https://www.petrikainulainen.net/software-development/processes/writing-just-enough-documentation/) to learn more about what to include in a developer guide.
```
Comment (wkurniawan07) [inline contents/projectManagement/documentation.md]: Add a break between "They should..." and "See this ...".

File: contents/projectManagement/documentation.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+A good README should convince users to use your product or contribute to your code, by providing enough information to make it easy for them to get started.
+For a good guide on how to write a README, check out [this guide](http://www.writethedocs.org/guide/writing/beginners-guide-to-docs/).
+Some suggest that READMEs should be written before you start coding, as it gives you an overview of your project and what you need to do.
+This is known as [Readme Driven Development](http://tom.preston-werner.com/2010/08/23/readme-driven-development.html).
+
+#### User Guide
+
+A user guide teaches users how to use your software.
+User guides can include quickstart guides, reference materials, tutorials, cookbooks/recipes and more.
+[Here](https://jacobian.org/writing/what-to-write/) is a useful guide on how to tackle different types of user documentation.
```
Comment (wkurniawan07) [inline contents/projectManagement/documentation.md]: Add a break between "User guides can..." and "Here is a useful guide...".

File: contents/projectManagement/documentation.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+Code docummentation refers to documentation that is found within the source code.
+All the following types of code documentation are important and complement one another.
+Note that such documentation should always be updated together with the code.
+
+#### Comments
+
+Comments are a well-known form of documentation.
+They [complement the code](http://softwareengineering.stackexchange.com/a/285789) and are intended for developers who use and work on the code.
+They include variable, method and class comments, as well as inline comments.
```
Comment (wkurniawan07) [inline contents/projectManagement/documentation.md]: Consider rephrasing to:

They include **API documentation** for variables, methods, classes, and packages, as well as **inline comments**.

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]
Great work in general! Good content and easy to follow.

You may also consider adding a section on documentation pitfalls/anti-patterns or how not to document. Of course it could be deferred to your next book chapter.

**Comment by kychua** [comment]
@wkurniawan07 Thanks for the comments! I've updated the chapter, will do the section on documentation pitfalls next time.

**Review by wkurniawan07 — APPROVED** [review]
LGTM!

File: contents/projectManagement/documentation.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+>
+> [Advice for Computer Science College Students](https://www.joelonsoftware.com/2005/01/02/advice-for-computer-science-college-students/), by Joel Spolsky
+
+## Why Write Documentation
+
+There are many benefits to writing good documentation, such as the following:
+
+* **Make things easier for yourself**: when you look back at your code a few months later, well-written documentation can make things a lot easier by explaining what you had been thinking when you were writing the code, when you find yourself confused by your own work
```
Comment (damithc) [inline contents/projectManagement/documentation.md]: Make or Makes?

File: contents/projectManagement/documentation.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+* For inspiration: https://github.com/PharkMillups/beautiful-docs
+* An overview of the different types of documentation and good examples of each type: http://www.ybrikman.com/writing/2014/05/05/you-are-what-you-document/
+* Documentation principles: http://www.writethedocs.org/guide/writing/docs-principles/
+* Agile documentation: http://www.agilemodeling.com/essays/agileDocumentationBestPractices.htm
+* Video on how to write great documentation: https://www.youtube.com/watch?v=z3fRu9pkuXE
+
+## Acknowledgements
+
+In addition to the links mentioned above, here are the links used in the writing of this book chapter:
```
Comment (damithc) [inline contents/projectManagement/documentation.md]: Instead of `book chapter`, use `document` or `article`

**Review by damithc — COMMENTED** [review]
Good work @kychua 
Some comments added. Revise as you see fit.
Adding more content and/or getting more peer reviews is welcome but not required. 
I will be merging end of week 12

**Pull request merged**